sdk/core/azure-core/inc/azure/core/http/policy.hpp

```diff
@@ -321,55 +321,25 @@ namespace Azure { namespace Core { namespace Http {
     void operator=(BearerTokenAuthenticationPolicy const&) = delete;
 
   public:
-    /**
-     * @brief Construct a Bearer Token authentication policy with single authentication scope.
-     *
-     * @param credential A #TokenCredential to use with this policy.
-     * @param scope Authentication scope.
-     */
-    explicit BearerTokenAuthenticationPolicy(
-        std::shared_ptr<TokenCredential const> credential,
-        std::string scope)
-        : m_credential(std::move(credential))
-    {
-      m_scopes.emplace_back(std::move(scope));
-    }
-
-    /**
-     * @brief Construct a Bearer Token authentication policy with multiple authentication scopes.
-     *
-     * @param credential A #TokenCredential to use with this policy.
-     * @param scopes A vector of authentication scopes.
-     */
-    explicit BearerTokenAuthenticationPolicy(
-        std::shared_ptr<TokenCredential const> credential,
-        std::vector<std::string> scopes)
-        : m_credential(std::move(credential)), m_scopes(std::move(scopes))
-    {
-    }
-
     /**
      * @brief Construct a Bearer Token authentication policy with multiple authentication scopes.
      *
-     * @tparam A type of scopes sequence iterator.
-     *
      * @param credential A #TokenCredential to use with this policy.
-     * @param scopesBegin An iterator pointing to begin of the sequence of scopes to use.
-     * @param scopesEnd An iterator pointing to an element after the last element in sequence of
-     * scopes to use.
+     * @param getTokenOptions #GetTokenOptions.
      */
-    template <typename ScopesIterator>
     explicit BearerTokenAuthenticationPolicy(
         std::shared_ptr<TokenCredential const> credential,
-        ScopesIterator const& scopesBegin,
-        ScopesIterator const& scopesEnd)
-        : m_credential(std::move(credential)), m_scopes(scopesBegin, scopesEnd)
+        GetTokenOptions const& getTokenOptions)
```

Comment:
```suggestion
        GetTokenOptions const& options)
```
Shouldn't this be optional?

Reply:
I don't think it is useful without scopes.
Optionality rules should not apply the same way - it is not options for the policy, these are options for GetToken() operation.